### PR TITLE
Fix flaky Dance Party UI test

### PR DESCRIPTION
**Pull request by fisher-alice**
This PR updates the `dance_party.feature` UI test to resolve the flakiness issue first reported [here](https://codedotorg.slack.com/archives/C0T0PNTM3/p1699578533080019) on Nov 9, 2023. Since this flakiness was blocking folks, https://github.com/code-dot-org/code-dot-org/pull/54876 skipped the problematic UI test scenario, and then https://github.com/code-dot-org/code-dot-org/pull/54874 updated it to check if the number of sprites displayed is `>= 10` instead of `=== 10`.
The issue was that when the project URL was in shared view, the number of sprites doubled (20 instead of 10).

doubled

In the Slack discussion, there was a suggestion that maybe we need to update the chrome version in testing environments. I tried updating to v109 [here](https://app.saucelabs.com/tests/1fe1f51791d742209b2c756f1eb52b37#51) but this did not resolve the doubling issue.

On the day the flakiness was first reported, https://github.com/code-dot-org/code-dot-org/pull/54792 was merged to `staging` which made updates to the `New Dance Lab Project.level` file including setting `uses_preview` to `true`.

In the 'Dance Party Share' scenario, the user first navigates to the project-backed level "http://studio.code.org/s/dance/lessons/1/levels/13". Then they copy the 'Share' url and navigate to the shared version of the project. It seems that since on the Dance Party level `uses_preview` is set to `false` while for Dance Party project  levels `uses_preview` is set to `true`, when the program is first run in 'share view' this incongruence sometimes results in the strange doubling of sprites. I am unsure why this happens in testing environments but is not able to be repro'd in other environments. Perhaps the call to [`window.__DanceTestInterface`](https://github.com/code-dot-org/code-dot-org/blob/a76eafdecf7e1054ae81805de0a9e3811150730e/apps/src/dance/Dance.js#L462) is a factor.

To fix the flakiness, I run and reset the program once. Then I run program again and test that the number of sprites displayed is correct.

## Links
[jira](https://codedotorg.atlassian.net/browse/LABS-241)
[Slack discussion](https://codedotorg.slack.com/archives/C0T0PNTM3/p1699578533080019)

## Testing story
Tested locally.

## Follow-up work
In /projects, Dance Party projects that have created sprites in the user program display thumbnail images of the preview of the project. If sprites are not created in the user program, then the default 'click to run' image is displayed.

Screenshot 2024-01-27 at 6 36 33 PM

Screenshot 2024-01-27 at 6 36 40 PM

 Now that we have the visualization column preview in Dance Party projects, I wonder if we should display the preview in ['share view'](https://github.com/code-dot-org/code-dot-org/blob/14169a2e5b32f111dd092de8011e6e183ee72c85/apps/src/dance/DanceVisualizationColumn.jsx#L111) instead of the default 'click to run' image. @samantha-code 

Screenshot 2024-01-27 at 6 38 48 PM

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/test/ui/features/star_labs/dance/dance_party.feature [hunk 1]
```diff
@@ -68,18 +68,20 @@ Feature: Dance Party
 
     When I navigate to the shared version of my project
     And element ".signInOrAgeDialog" is hidden
+    # Tested level from Dance Party 2018 level 13 does not use visualization column preview.
```
Comment (bencodeorg) [inline dashboard/test/ui/features/star_labs/dance/dance_party.feature]: Nit: it's not the "checking" of run/reset buttons that resolves flakiness, right? It's running it once gets us into a consistent state? Something like:

```
We run/reset here once before checking the number of sprites below,
which helps resolve some not fully understood test flakiness related to Dance Party's "preview mode".
The level used in this test (Dance 2018, level 13) does not use preview mode,
[but sometimes ends up in preview mode sometimes when shared I don't understand this part :)]
```

**Review by bencodeorg — APPROVED** [review]
Nice sleuthing!

I think reading thru this points out a lack of understanding on my part re: project sharing from a level...so when you share a project from a Dance Party level and someone else opens it, they're opening it in the context of a different level (the Dance standalone project level)?

**Review by thomasoniii — APPROVED** [review]
lgtm 🚀

**Comment by fisher-alice** [comment]
> Nice sleuthing!
> 
> I think reading thru this points out a lack of understanding on my part re: project sharing from a level...so when you share a project from a Dance Party level and someone else opens it, they're opening it in the context of a different level (the Dance standalone project level)?

My understanding is that when you [click on the 'Share' button](https://github.com/code-dot-org/code-dot-org/blob/16970a535ee9adcc5567a23f3fd777f4e75fd236/apps/src/code-studio/components/header/ProjectShare.jsx#L20) of a [channel-backed level](https://github.com/code-dot-org/code-dot-org/blob/16970a535ee9adcc5567a23f3fd777f4e75fd236/dashboard/app/models/levels/level.rb#L402) (free-play Dance levels are channel-backed), you get the project share URL which uses the Dance standalone project level.

File: dashboard/test/ui/features/star_labs/dance/dance_party.feature (hunk 1 above)
Comment (fisher-alice) [reply dashboard/test/ui/features/star_labs/dance/dance_party.feature]: @bencodeorg - How does this sound?
```
# We run/reset here once before checking the number of sprites below which helps
# resolve some not fully understood test flakiness related to Dance Party's "preview mode".
# The level used in this test (Dance Party 2018, level 13) does not use preview mode but
# when the level is shared via `project.getShareUrl`, the standalone Dance project level
# does use preview mode.

```

**Comment by breville** [comment]
> I am unsure why this happens in testing environments but is not able to be repro'd in other environments. Perhaps the call to [window.__DanceTestInterface](https://github.com/code-dot-org/code-dot-org/blob/a76eafdecf7e1054ae81805de0a9e3811150730e/apps/src/dance/Dance.js#L462) is a factor.

It's concerning that we weren't able to repro this.  If we suspect the call to `window.__DanceTestInterface` is a factor, we could open a developer console and run it at the appropriate step in a manual repro attempt.  Happy to pair on this investigation if you'd like to dig deeper.

**Pull request merged**